File: src/main/java/de/dfki/mary/coefficientextraction/extraction/ExtractBase.java
package de.dfki.mary.coefficientextraction.extraction;

import java.io.BufferedReader;
import java.io.File;
import java.io.InputStreamReader;
import java.util.Hashtable;

/**
 * Interface to provide an coefficient extraction method
 *
 * @author <a href="mailto:slemaguer@coli.uni-saarland.de">slemaguer</a>
 */
public abstract class ExtractBase implements ExtractInterface
{
    protected Hashtable<String, File> extToFile;

    public void setOutputFiles(Hashtable<String, File> extToFile)
    {
        this.extToFile = extToFile;
    }

    public abstract void extract(File input_file) throws Exception;

    public void run(String command) throws Exception {
        // Run extraction
        String[] cmd = {"bash", "-c", command};
        Process p = Runtime.getRuntime().exec(cmd);
        p.waitFor();

        if (p.exitValue() != 0) {
            throw new Exception("Command failed: " + command);
        }
    }
}

File: src/main/kotlin/de/dfki/mary/coefficientextraction/extraction/ExtractBase.kt
package de.dfki.mary.coefficientextraction.extraction;

// Logging
import org.apache.log4j.Logger

// File
import java.io.File

import kotlin.collections.*

/**
 * Interface to provide an coefficient extraction method
 *
 * @author <a href="mailto:slemaguer@coli.uni-saarland.de">slemaguer</a>
 */
abstract class ExtractBase: ExtractInterface
{
    @JvmField protected val logger: Logger = Logger.getLogger(ExtractBase::class.java.name)
    @JvmField protected var extToFile: Map<String, File> = emptyMap<String, File>();

    override fun setOutputFiles(extToFile: Map<String, File>) {
        this.extToFile = extToFile;
    }


    // override abstract fun extract(input_file: File);

    fun run(command: String) {
        var process = ProcessBuilder(listOf("sh", "-c", command))
            .directory(File(System.getProperty("user.dir")))
            .redirectErrorStream(true)
            .start()

        process.inputStream.reader().forEachLine { out -> logger.info(out) }
        process.waitFor();

        if (process.exitValue() != 0) {
            throw Exception("hts command failed: " + command)
        }
    }
}
